src/main/java/org/codehaus/mojo/extraenforcer/dependencies/EnforceManagedDepsRule.java

```diff
@@ -0,0 +1,158 @@
+package org.codehaus.mojo.extraenforcer.dependencies;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.maven.enforcer.rule.api.AbstractEnforcerRule;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.ModelBase;
+import org.apache.maven.model.Profile;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Enforcer rule that will check that all dependencies are managed.
+ */
+@Named("requireManagedDeps")
+public class EnforceManagedDepsRule extends AbstractEnforcerRule {
+    private boolean checkProfiles = true;
+
+    private boolean failOnViolation = true;
+
+    private Pattern[] regexIgnoredPatterns;
+
+    private final MavenProject project;
+
+    @Inject
+    public EnforceManagedDepsRule(final MavenProject project) {
+        this.project = Objects.requireNonNull(project);
+    }
+
+    @Override
+    public void execute() throws EnforcerRuleException {
+
+        final Model model = project.getOriginalModel();
+
+        final Set<Dependency> failed = new HashSet<Dependency>();
+
+        getLog().debug("Checking model...");
+
+        check(model, failed);
+
+        if (checkProfiles) {
+            getLog().debug("Checking profiles...");
+            final List<Profile> profiles = project.getOriginalModel().getProfiles();
```

Comment:
@slawekjaranowski In the original project this line was:
```
final List<Profile> profiles = project.getModel().getProfiles();
```
But this failed with maven-3.9.11 when I tested it.
Change it to:
```
final List<Profile> profiles = project.getOriginalModel().getProfiles();
```
makes the test pass with maven-3.9.11 but it fails with maven-3.6.3.

I do not understand enough of maven internals to get a working solution for old 3.6.3 and new 3.9.11 ... Can you suggest how I can solve this issue? Or is this check completely wrong?